CHAT_SERVER/chat_server.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bufio"
+	"net"
+	"strings"
+)
+
+const (
+	CONN_TYPE = "tcp"
+)
+
+type Client struct {
+	id       string
+	name     string
+	incoming chan string
+	reader   *bufio.Reader
```

Comment:
Why you use `bufio.Reder` and `bufio.Writer`? `net.Conn` that you got from `listener.Accept()` already implements both of them =)